lib/cli/scripts/github-folder-downloader.ts

```diff
@@ -0,0 +1,51 @@
+import axios, { AxiosResponse } from 'axios';
+import AdmZip from 'adm-zip';
+import path from 'path';
+
+export class GitHubRepoUtils {
+    private baseURL = 'https://api.github.com';
+    private token: string;
+    private repo: string;
+    private subFolderName: string;
+
+    constructor(token: string, repo: string) {
+        this.token = token;
+        this.repo = repo;
+    }
+
+    async downloadRepoAndUnpackRepository(owner: string, branch = 'main'): Promise<void> {
```

Comment:
This doesn't sound right 😅 
```suggestion
    async downloadAndUnpackRepository(owner: string, branch = 'main'): Promise<void> {
```